rust/noosphere-gateway/src/again.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::Result;
+use std::future::Future;
+use std::sync::Arc;
+use tokio::sync::OnceCell;
+
+/// Wraps an "initialization" step, expressed as a closure, and allows
+/// a user to invoke closures with the result of that initialization step
+/// as a context argument. The result of invocation is always returned, but
+/// a failure result causes the initialized context to be reset is that it
+/// is re-initialized upon the next invocation attempt.
```

Comment:
Last sentence might be missing a word -- could be a good ASCII FSM diagram candidate